solidity/test/Polymer7683.t.sol

```diff
@@ -0,0 +1,392 @@
+        
+        bytes32 orderId = bytes32("orderId1");
+        bytes memory fillerData = abi.encode(bytes32("filler1"));
+        
+        // Create proof for wrong chain ID
+        bytes memory proof = _createSettlementProof(999, destContract, orderId, fillerData);
+        
+        vm.expectRevert(Polymer7683.UnregisteredDestinationChain.selector);
+        polymer7683.handleSettlementWithProof(proof);
+    }
+
+    function test_handleSettlementWithProof_unregisteredDestination() public {
+        bytes32 orderId = bytes32("orderId1");
+        bytes memory fillerData = abi.encode(bytes32("filler1"));
+        
+        bytes memory proof = _createSettlementProof(destChainId, destContract, orderId, fillerData);
+        
+        vm.expectRevert(Polymer7683.UnregisteredDestinationChain.selector);
+        polymer7683.handleSettlementWithProof(proof);
+    }
+
+    function test_handleSettlementWithProof_invalidOrderId() public {
+        vm.prank(owner);
+        polymer7683.setDestinationContract(destChainId, destContract);
+        
+        // Create proof for one order ID
+        bytes32 proofOrderId = bytes32("order1");
+        bytes memory fillerData = abi.encode(bytes32("filler1"));
+
+        console2.log("\nTest setup:");
+        console2.log("Proof Order ID:");
+        console2.logBytes32(proofOrderId);
+        console2.log("Submitted Order ID:");
+        console2.logBytes32(bytes32("order2"));
+        
+        bytes memory proof = _createSettlementProof(destChainId, destContract, proofOrderId, fillerData);
+
+        (, , bytes memory topics, bytes memory eventData) = prover.validateEvent(proof);
+
+        console2.log("\nDecoding event data...");
+        (
+            bytes32 decodedOrderId,
+            bytes memory decodedOriginData,
+            bytes memory decodedFillerData
+        ) = abi.decode(eventData, (bytes32, bytes, bytes));
+        console2.log("Decoded Order ID from proof:");
+        console2.logBytes32(decodedOrderId);
+        console2.log("Decoded Origin Data length:", decodedOriginData.length);
+        console2.log("Decoded Filler Data length:", decodedFillerData.length);
+      
+        // Try to submit different order ID
+        bytes32 submittedOrderId = bytes32("order2");
+        
+        vm.expectRevert(Polymer7683.InvalidEventData.selector);
+        try polymer7683.handleSettlementWithProof(proof) {
+            console2.log("Call succeeded unexpectedly");
+        } catch Error(string memory reason) {
+            console2.log("Reverted with reason:", reason);
+        } catch (bytes memory errData) {
+            console2.log("Reverted with raw error data length:", errData.length);
+            if (errData.length >= 4) {
+                bytes4 selector = bytes4(errData);
+                console2.log("Error selector:");
+                console2.logBytes4(selector);
+            }
+        }
+    }
+
+    function test_handleRefundWithProof_success() public {
```

Comment:
I suspect this test is not working as expected, because the order hasn't been open
I would assert that the `_handleRefundOrder` works. Provably asserting that the `Refunded` is emitted.

Reply:
updated the test to open an order first to be more realistic.